### DISCUSSION TITLE
umi4在配置mf时候 如何把整个project作为一个expose

### DISCUSSION THREAD

**Discussion by ghost** [umijs/umi#10046] [Q&A]
查看文档  现在exposes是一个约定文件夹  我想把整个project作为一个expose导出去 好像没有找到方法

**Comment by ghost** [comment]
@stormslowly 大佬

**Comment by stormslowly** [comment]
整个项目  就直接用 qiankun 微应用的形式好了

mf 还是在模块级别的共享

**Reply by ghost** [reply]
主要还是想用webpack原生的module federation 功能

**Reply by stormslowly** [reply]
而且导出整个 umi 应用 应用没有一个模块定义，目前 .umi/umi.ts 也没有导出“模块”的定义

**Reply by ghost** [reply]
@stormslowly  以后会不会支持 不支持的话其实 umi 是用不了原生 module federation 的功能的

**Reply by stormslowly** [reply]
你可以这么试试看 
用 plugin.ts 搞个插件，复制一份 umi.ts 里面的 render 实现导出，然后把这个方法再 expose 里定义导出试试

**Reply by stormslowly** [reply]
想问下 一定要用 mf 的原因是啥啊？

**Reply by ghost** [reply]
公司自己设计的mFE 架构,他是基于mf 设计的, 但是自定义的hook 与 qiankun 的hook 又不一样,所以不能直接用qiankun ,需要他们的hook 里面 进行 
```
const rootDom = reactDom.createRoot(dyamnicElement)
rootDom.render(<UmiApp>)


```

**Reply by ghost** [reply]
<img width="1239" alt="image" src="https://user-images.githubusercontent.com/15773714/207759175-cae6fc24-f68e-4f56-84e0-5b2f126c4a7d.png">
乾坤也有这样的api 规范,这在umi qiankun plugin 怎么实现的. 现在umi 封装的等于只能接入qiankun mfe ,其他的接入不了

**Reply by ghost** [reply]
我看 qiankun 插件暴露出来一个 microApp  的 东西 是不是就是 等于 ReactDon.render 里的整个umi app,如果是这样的话,希望microApp 可以一直暴露出来, 然后可以 作为 mf 的expose
